src/main/java/ro/msg/learning/shop/controller/ExportStockController.java

```diff
@@ -0,0 +1,25 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.learning.shop.dto.StockDTO;
+import ro.msg.learning.shop.services.ExportStockServices;
+
+import java.util.List;
+
+@RestController
+public class ExportStockController {
+    private final ExportStockServices stockServices;
+
+    public ExportStockController(ExportStockServices stockServices) {
+        this.stockServices = stockServices;
+    }
+
+    @RequestMapping(value = "/exportStock/{id}", method = RequestMethod.GET)
```

Comment:
You can use `@GetMapping` instead - Spring has specialized `@...Mapping` annotations for doing `@RequestMapping`s for a specific HTTP method: `@GetMapping`, `@PostMapping`, etc.

Reply:
Also, the URL here is not REST-like. Instead, I would do a mapping like `POST /locations/{id}/stocks/export`.